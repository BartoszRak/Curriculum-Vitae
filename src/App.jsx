import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { BrowserRouter } from 'react-router-dom'
import { withStyles } from '@material-ui/core'
import { addLocaleData, IntlProvider } from 'react-intl'
import { flatten } from 'flat'
import localeEn from 'react-intl/locale-data/en'
import localePl from 'react-intl/locale-data/pl'

import withMuiTheme from '~theme/withMuiTheme'
import Layout from './layout'
import Loader from '~components/Loader'
import PreventInternetExplorer from '~components/PreventInternetExplorer'
import { usePayloadLoading, useKeyboardCapture } from '~hooks'
import LanguageContext from '~services/Internacionalization'
import ApplicationContext from '~services/AppState'
import messagesPl from '~services/Internacionalization/translations/pl.json'
import messagesEn from '~services/Internacionalization/translations/en.json'

import style from './App.style'

const messages = {
  en: flatten(messagesEn),
  pl: flatten(messagesPl),
}
addLocaleData([...localeEn, ...localePl])
export function App({ classes }) {
  const { loading } = usePayloadLoading()
  const [language, setLanguage] = useState('en')
  const [mode, setMode] = useState('normal')

  useKeyboardCapture([
    command => {
      if (command === 'snake') setMode('play')
      if (command === 'stop') setMode('normal')
    },
  ])

  return (
    <BrowserRouter>
      <LanguageContext.Provider
        value={{
          language,
          setLanguage,
        }}
      >
        <LanguageContext.Consumer>
          {({ language: chosenLanguage }) => (
            <IntlProvider
              key={chosenLanguage}
              locale={chosenLanguage}
              messages={messages[chosenLanguage]}
            >
              <Loader className={classes.loader} disabled={!loading}>
                <PreventInternetExplorer>
                  <ApplicationContext.Provider value={{ mode, setMode }}>
                    {mode === 'normal' && <Layout />}
                    {mode === 'play' && <div>lets play</div>}
                  </ApplicationContext.Provider>
                </PreventInternetExplorer>
              </Loader>
            </IntlProvider>
          )}
        </LanguageContext.Consumer>
      </LanguageContext.Provider>
    </BrowserRouter>
  )
}

App.propTypes = {
  classes: PropTypes.object,
}

export default withMuiTheme(withStyles(style)(App), true)
