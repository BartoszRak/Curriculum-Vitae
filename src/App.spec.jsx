import React from 'react'
import { shallow } from 'enzyme'

import { App } from './App'

jest.mock('./layout', () => 'LayoutMock')
jest.mock('~components/PreventInternetExplorer', () => 'PreventInternetExplorerMock')

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<App {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: App', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
})
