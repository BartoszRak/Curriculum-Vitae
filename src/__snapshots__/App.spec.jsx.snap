// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: App rendering match snapshot 1`] = `
<BrowserRouterMock>
  <ContextProvider
    value={
      Object {
        "language": "en",
        "setLanguage": [Function],
      }
    }
  >
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </ContextProvider>
</BrowserRouterMock>
`;
